### DISCUSSION TITLE
Supporting google/budoux for better line breakings

### DISCUSSION THREAD

**Discussion by peacock0803sz** [slidevjs/slidev#1839] [Ideas]
# Problem and Motivation

In some multibite Languages, It's too hard to make line breaklings correctly 😕 
I'd like to use [Budoux (JavaScript module)](https://github.com/google/budoux/tree/main/javascript/) to solve this problem, but I felt configuring it as a vite plugin is pretty difficult...
Budoux is the machine learning powered line break organizer tool, and it's standalone, small, language-neutral.

How can I apply Bodoux for slidev slides?
Just installing [budoux npm package](https://www.npmjs.com/package/budoux) and adding `import` to somewhere TypeScript file, It's not working...

# Actual / Expected Behaviors

![CleanShot 20240824-154936](https://github.com/user-attachments/assets/b7678fd6-f6a3-4af5-9c55-977979e24d1c)

**Comment by ryoppippi** [comment]
So, I created a transformer to apply budoux to statements in slide.

https://github.com/ryoppippi/slidev-transformer-budoux

This makes your life easier.

**Reply by peacock0803sz** [reply]
@ryoppippi Thanks, looks so cool! Can I use via custom HTML class or CSS property (better)?

**Reply by ryoppippi** [reply]
What do you mean "custom html"? Give me an example

**Reply by peacock0803sz** [reply]
Oh, I meant "Custom class attributes" like this

```html
<div class="bodoux">
Here to something better line breaking
</div>
```

Or, via using`@apply` in CSS:
```css
.slidev-layout.default div {
  @apply budoux
}
```

**Reply by peacock0803sz** [reply]
@ryoppippi It's a just a kindly reminder) Is there any updates or ideas? I hope to be able to use without `<budoux>` tag elements 👀

**Reply by ryoppippi** [reply]
Based on our personal conversation with @peacock0803sz , I cannot implement it because `slidev transformer` does not process DOM, so adding `class` does not work
